src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs

```diff
@@ -119,13 +119,17 @@ int IVsSolutionLoadEvents.OnAfterBackgroundSolutionLoadComplete()
 
             OutputToOutputWindow($"Getting project information - start");
             var start = DateTimeOffset.UtcNow;
-            // Capture the context so that we come back on the UI thread, and do the actual project creation there.
-            var projectInfos = await deferredProjectWorkspaceService.GetDeferredProjectInfoForConfigurationAsync(
-                $"{solutionConfig.Name}|{solutionConfig.PlatformName}",
-                cancellationToken).ConfigureAwait(true);
-            AssertIsForeground();
+            var projectInfos = SpecializedCollections.EmptyReadOnlyDictionary<string, DeferredProjectInformation>();
+            if (solutionConfig != null)
+            {
+                // Capture the context so that we come back on the UI thread, and do the actual project creation there.
+                projectInfos = await deferredProjectWorkspaceService.GetDeferredProjectInfoForConfigurationAsync(
+                    $"{solutionConfig.Name}|{solutionConfig.PlatformName}",
+                    cancellationToken).ConfigureAwait(true);
+                AssertIsForeground();
```

Comment:
I'd prefer we put that outside the if block, so that way there's no chance we end up on different threads if the conditional is taken vs. not taken due to future edits here.